### PR TITLE
Error status code and message

### PR DESCRIPTION
**Pull request by jbghoul**
* Provides status code and message on thrown errors
* Complete unit test
* Documentation

### CODE REVIEW

File: test/http-request.js [hunk 1]
```diff
@@ -233,7 +247,7 @@ describe("http-request", function() {
                     throw new Error("ShouldNotBeCalled");
                 })
                 .catch(function(error) {
-                    expect(error).to.match(/HttpConnectionError/);
+                    expect(error).to.match(/HttpConnexionError/);
```
Comment (flozz) [inline test/http-request.js]: Why did you change the matched error? `Connexion` is a French word, so it should not be used as error name.

Comment (jbghoul) [reply test/http-request.js]: I don't remember editing this consciously. Fixed

**Comment by flozz** [comment]
Released in **v1.4.0**
